subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheEncryptionIntegrationTest.groovy

```diff
@@ -160,109 +143,96 @@ class ConfigurationCacheEncryptionIntegrationTest extends AbstractConfigurationC
         }
     }
 
-    def "new configuration cache entry if was encrypted but encryption is off"() {
+    def "build fails if keystore password is incorrect"() {
         given:
-        def configurationCache = newConfigurationCacheFixture()
-        runWithEncryption(true)
+        runWithEncryption()
 
         when:
-        runWithEncryption(false)
+        keyStorePassword = "foobar"
+        fails(*(["help", "--configuration-cache"] + encryptionOptions))
 
         then:
-        configurationCache.assertStateStored()
-        outputContains("Calculating task graph as no configuration cache is available for tasks: help")
+        result.assertHasErrorOutput """
+Error loading encryption key from Java keystore at ${keyStorePath}
+> Integrity check failed: java.security.UnrecoverableKeyException: Failed PKCS12 integrity checking
```

Comment:
❓ What should users do in this case?

Reply:
Deleting the keystore would make it go away